rest/src/main/kotlin/rys/rest/service/impl/ChatServiceImpl.kt

```diff
@@ -1,35 +1,39 @@
-package rys.ajaxpetproject.service.impl
+package rys.rest.service.impl
 
 
 import org.bson.types.ObjectId
 import org.springframework.stereotype.Service
-import rys.ajaxpetproject.exception.ChatNotFoundException
-import rys.ajaxpetproject.model.MongoChat
-import rys.ajaxpetproject.repository.ChatRepository
-import rys.ajaxpetproject.service.ChatService
+import rys.rest.exceptions.ChatNotFoundException
+import rys.rest.model.MongoChat
+import rys.rest.repository.ChatRepository
+import rys.rest.service.ChatService
 
 @Service
 class ChatServiceImpl(val chatRepository: ChatRepository): ChatService {
 
-    override fun createChat(mongoChat: MongoChat): MongoChat = chatRepository.save(mongoChat)
+    override fun createChat(mongoChat: MongoChat): MongoChat {
+        return chatRepository.save(mongoChat)
+    }
 
     override fun findChatById(id: ObjectId): MongoChat? = chatRepository.findChatById(id)
 
-    fun getChatById(id: ObjectId): MongoChat = chatRepository.findChatById(id) ?: throw ChatNotFoundException()
-
-    fun getAllChats(): List<MongoChat> = chatRepository.findAllBy()
-
     override fun findAllChats(): List<MongoChat> = chatRepository.findAllBy()
 
-    override fun updateChat(id: ObjectId, updatedMongoChat: MongoChat): MongoChat =
+    override fun updateChat(id: ObjectId, updatedMongoChat: MongoChat): MongoChat {
         findChatById(id)
-            .let { chatRepository.save(updatedMongoChat) }
+            .let {
+                val newChat = updatedMongoChat.copy(id = id)
+                chatRepository.save(newChat)
+            }
+        return  findChatById(id)!!
```

Comment:
it's not clear why:
1. you invoke findChatById twice
2. why you have changed the function body at all